From 9674316bf33c6fbf82110a8757253980f03cd78b Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 29 Sep 2017 10:55:03 +0800
Subject: [PATCH 2574/5242] MLK-16551 ARM64: dts: fsl-imx8qxp-mek: enable USB2
 port

commit  cfffa52a7118eac0de432583d5f4e66c5ae734c7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Both host and device mode are support, and the port is at base board.

Below rework is needed:
Remove R145, R143, R1390, and install R144, R142, R1389.

BuildInfo:
- SCFW 1f59442e, IMX-MKIMAGE fb52c576, ATF
- U-Boot 2017.03-imx_v2017.03+g34be5a2

Acked-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index f4c0774..9c51e74 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -55,6 +55,16 @@
 			gpio = <&pca9557_a 0 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_usb_otg1_vbus: regulator@0 {
+			compatible = "regulator-fixed";
+			reg = <0>;
+			regulator-name = "usb_otg1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&pca9557_b 2 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 	};
 
 	sound: sound {
@@ -510,6 +520,16 @@
 	status = "okay";
 };
 
+&usbotg1 {
+	vbus-supply = <&reg_usb_otg1_vbus>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	power-polarity-active-high;
+	disable-over-current;
+	status = "okay";
+};
+
 /* Only enable it and ptn5110 after rework U182 */
 &usbotg3 {
 	dr_mode = "otg";
-- 
1.7.9.5

